tests/testthat/test-path.R

```diff
@@ -61,6 +61,17 @@ test_that("split_path_ext works", {
   )
 })
 
+test_that("vroom() informs user to use I() for literal data (or not)", {
+  expect_snapshot(
+    x <- vroom("a,b,c,d\n1,2,3,4", show_col_types = FALSE)
+  )
+
+  f <- function(file) vroom(file, show_col_types = FALSE)
+  expect_snapshot(
+    x <- f("a,b,c,d\n1,2,3,4")
+  )
```

Comment:
I expected this scenario to _not_ warn the user 🤔

Reply:
Because it's called from a function? But that function inherits either from global (if you run the test interactively) or from your package namespace (if you run the test through `test()` [load-all'd namespace] or `check()` [cloned namespace]). So it counts as a direct usage.

If you want to reassure yourself of the indirect usage, you'll have to construct a function that inherits from another namespace like rlang or utils. We do this in the lifecycle tests but I wouldn't expect a package to do such tests. It's a bit tricky because you can't rely on the lexical scope of the unit test and instead ns-qualify everything and pass all data by argument or by injection. For this simple function though, that's only a matter of adding:

```r
environment(f) <- ns_env("utils")
```

Then this will count as an indirect usage because the warning will be emitted from a package that is not being tested.